src/components/TaskCompleter.js

```diff
@@ -0,0 +1,72 @@
+import React from "react";
+
+class TaskCompleter extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      timeTaken: "",
+      keyTakeaways: "",
+    };
+  }
+
+  handleSubmit = (e) => {
+    e.preventDefault();
+
+    const { tasksCompleted, id, reflectionToDo } = this.props;
+    const { timeTaken, keyTakeaways } = this.state;
+
+    const index = tasksCompleted.findIndex((task) => task.id === id); // loops through the array of objects and returns the index of the matching task
+
+    if (index !== -1) {
+      reflectionToDo(index, timeTaken, keyTakeaways);
+      tasksCompleted[index].timeTaken = timeTaken;
+      tasksCompleted[index].keyTakeaways = keyTakeaways;
```

Comment:
Here you are manually updating a task within the state. This should be avoided as it might lead to unpredictable behaviour and bugs. You should use your state update function for this.